cosmwasm/deploy-contract.js

```diff
@@ -0,0 +1,404 @@
+        }
+    }
+
+    throw new Error(`${contractName} is not supported.`);
+};
+
+const deploy = async (options, chain, config) => {
+    printInfo('Deploying for chain', chain ? chain.name : 'none');
+
+    const wallet = await DirectSecp256k1HdWallet.fromMnemonic(options.mnemonic, { prefix: 'axelar' });
+    const client = await SigningCosmWasmClient.connectWithSigner(config.axelar.rpc, wallet);
+
+    if (config.axelar.contracts[options.contractName] === undefined) {
+        config.axelar.contracts[options.contractName] = {};
+    }
+
+    const contractConfig = config.axelar.contracts[options.contractName];
+
+    printInfo('Contract name', options.contractName);
+
+    const reuseCodeId = !!options.reuseCodeId && !!contractConfig.codeId;
+    printInfo('Reusing codeId', reuseCodeId.toString());
+
+    if (!reuseCodeId) {
+        const result = await uploadContract(config, options, wallet, client);
+        contractConfig.codeId = result.codeId;
+
+        if (result.address) {
+            contractConfig.address = result.address;
+            printInfo('Expected contract address', contractConfig.address);
+        }
+    }
+
+    printInfo('Code Id', contractConfig.codeId);
+
+    if (!options.uploadOnly) {
+        const initMsg = makeInstantiateMsg(options.contractName, config, chain);
+        const contractAddress = await instantiateContract(config, options, options.contractName, initMsg, wallet, client);
+
+        if (chain) {
+            contractConfig[chain.id] = {
+                ...contractConfig[chain.id],
+                address: contractAddress,
+            };
+        } else {
+            contractConfig.address = contractAddress;
+        }
+
+        printInfo('Contract address', contractAddress);
+    }
+};
+
+const main = async (options) => {
+    const config = loadConfig(options.env);
+
+    let chains = options.chainNames.split(',').map((str) => str.trim());
+
+    if (options.chainNames === 'all') {
+        chains = Object.keys(config.chains);
+    }
+
+    for (const chain of chains) {
+        if (config.chains[chain.toLowerCase()] === undefined && chain !== 'none') {
+            throw new Error(`Chain ${chain} is not defined in the info file`);
+        }
+    }
+
+    for (const chain of chains) {
+        await deploy(options, config.chains[chain.toLowerCase()], config);
+        saveConfig(config, options.env);
+
+        options.reuseCodeId = true;
+    }
+};
+
+const programHandler = () => {
+    const program = new Command();
+
+    program.name('upload-contract').description('Upload CosmWasm contracts');
+
+    program.addOption(
+        new Option('-e, --env <env>', 'environment')
+            .choices(['local', 'devnet', 'stagenet', 'testnet', 'mainnet'])
+            .default('testnet')
+            .makeOptionMandatory(true)
+            .env('ENV'),
+    );
+    program.addOption(new Option('-m, --mnemonic <mnemonic>', 'mnemonic').makeOptionMandatory(true).env('MNEMONIC'));
```

Comment:
if it is decided that this script should wait for user confirmation, than you should also add an `-y/--yes` flag to override confirmation. Particular useful for devnet deployment.

Reply:
@cjcobb23 address this before I approve